roles/ec2_networking_resources/tasks/create.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Validate options
+  ansible.builtin.fail:
+    msg: "When creating resources, all of the following options must be provided: ec2_networking_resources_vpc_cidr_block, ec2_networking_resources_subnet_cidr_block, ec2_networking_resources_sg_internal_name"
+  when: ec2_networking_resources_vpc_cidr_block | default("", true) == "" or
+    ec2_networking_resources_subnet_cidr_block | default("", true) == "" or
+    ec2_networking_resources_sg_internal_name | default("", true) == ""
+
+- name: Create VPC
+  amazon.aws.ec2_vpc_net:
+    name: "{{ ec2_networking_resources_vpc_name }}"
+    cidr_block: "{{ ec2_networking_resources_vpc_cidr_block }}"
+  register: ec2_networking_resources_vpc_result
+
+- name: Set VPC ID
+  ansible.builtin.set_fact:
+    vpc_id: "{{ ec2_networking_resources_vpc_result.vpc.id }}"
+
+- name: Create VPC subnet
+  amazon.aws.ec2_vpc_subnet:
+    vpc_id: "{{ ec2_networking_resources_vpc_result.vpc.id }}"
+    cidr: "{{ ec2_networking_resources_subnet_cidr_block }}"
+  register: ec2_networking_resources_subnet_result
+
+- name: Set subnet ID
+  ansible.builtin.set_fact:
+    subnet_id: "{{ ec2_networking_resources_subnet_result.subnet.id }}"
+
+- name: Create custom route table for subnet
```

Comment:
This route table is needed only when the user wants to create an internet gateway, I think this block can be moved to the line 66 with the creation of the internet gateway